File: src/main/java/in/dragonbra/javasteam/util/CollectionUtils.java
package in.dragonbra.javasteam.util;

import java.util.Map;
import java.util.Objects;

/**
 * @author lngtr
 * @since 2018-02-19
 */
public class CollectionUtils {
    public static <T, E> T getKeyByValue(Map<T, E> map, E value) {
        for (Map.Entry<T, E> entry : map.entrySet()) {
            if (Objects.equals(value, entry.getValue())) {
                return entry.getKey();
            }
        }
        return null;
    }
}

File: src/main/java/in/dragonbra/javasteam/util/CollectionUtils.kt
package `in`.dragonbra.javasteam.util

import `in`.dragonbra.javasteam.util.compat.ObjectsCompat

/**
 * @author lngtr
 * @since 2018-02-19
 */
object CollectionUtils {
    @JvmStatic
    fun <T, E> getKeyByValue(map: Map<T, E>, value: E): T? = map.entries.firstOrNull {
        ObjectsCompat.equals(value, it.value)
    }?.key
}
